src/aqua/spell/api.aqua

```diff
@@ -4,12 +4,12 @@ import "@fluencelabs/aqua-lib/builtin.aqua"
 import TriggerConfig from "spell_service.aqua"
 import SpellID from "types.aqua"
 
+service PeerSpellExt("spell"):
+    install(script: string, init_args: ⊤, config: TriggerConfig, alias: ?string) -> SpellID
```

Comment:
This looks a little weird to me. Maybe update existing `PeerSpell.install`?

Also, why is `alias` optional? Is there a use case for creating a spell without an alias?

Reply:
I don't want to make `alias` not optional 'til the IS tests are ready

We don't have logic to handle the situation when the alias setting fails, but at least atm we don't remove the spell when it fails. Simply using this function will change the current semantics.

Reply:
Well, isn't it good not to have an orphan spell without an alias? eliminate such situation sounds nice to me

Having two definitions of the same function sounds complex, and having one of the parameters optional just increases the complexity further 🤔

We can have `alias` optional on Nox level, for backward compatibility, but in aqua-lib I don't see why keep it backward  compatible.

Reply:
I changed the api so now we have only one function with an optional alias

But I think we should make the change in smaller steps.

Reply:
I still don't see why keep alias optional

Reply:
Because 
```
PeerSpell.install()
Srv.add_alias(alias)
```
and
```
PeerSpell.install(, alias)
```
behaves differently on errors.

And that's how `PeerSpell.install` it's used atm in the IS.

So, changing `PeerSpell.install() -> PeerSpell.install(, alias = nil)` won't break anything.

I think we need to put more thought into using `install` with non-optional alias in IS. For example, in a separate task to make IS better react on errors.

Reply:
If we change `PeerSpell.install` to `PeerSpell.install(, alias = "worker-spell")`, what will it break?

Reply:
It won't break on a happy path. It will behave differently on errors.

Reply:
How will it behave? 

It seems to me that we're discussing how to avoid the change, but it may be cheaper to just embrace the change completely.

Reply:
When it's impossible to set the alias for some reason, the spell will be removed in the new `PeerSpell.install`. In the current IS implementation, it won't be removed. I don't know how IS reacts to it. I don't feel that it's okay just to change and hope that everything will be okay.

Reply:
Let's change thoughtfully. Consider the code, and how it will change it. We will have to do that anyway, might skip transition-level APIs.

Reply:
I would rather do it in another task.

Reply:
okay, changed to non-optional. i guess, Loysha soon will write the tests and will find out if there are any errors.

Reply:
Thank you!